Documentation/kube-flannel.yml

```diff
@@ -1,9 +1,60 @@
 ---
+apiVersion: extensions/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: psp.flannel.unprivileged
+  annotations:
+    seccomp.security.alpha.kubernetes.io/allowedProfileNames: docker/default
+    seccomp.security.alpha.kubernetes.io/defaultProfileName: docker/default
+    apparmor.security.beta.kubernetes.io/allowedProfileNames: runtime/default
+    apparmor.security.beta.kubernetes.io/defaultProfileName: runtime/default
+spec:
+  privileged: false
+  volumes:
+    - configMap
+    - secret
+    - emptyDir
+    - hostPath
+  allowedHostPaths:
+    - pathPrefix: "/etc/cni/net.d"
+    - pathPrefix: "/etc/kube-flannel"
+    - pathPrefix: "/run"
+  readOnlyRootFilesystem: false
+  # Users and groups
+  runAsUser:
+    rule: RunAsAny
+  supplementalGroups:
+    rule: RunAsAny
+  fsGroup:
+    rule: RunAsAny
+  # Privilege Escalation
+  allowPrivilegeEscalation: false
+  defaultAllowPrivilegeEscalation: false
+  # Capabilities
+  allowedCapabilities: ['NET_ADMIN']
+  defaultAddCapabilities: []
+  requiredDropCapabilities: []
+  # Host namespaces
+  hostPID: false
+  hostIPC: false
+  hostNetwork: true
+  hostPorts:
+  - min: 0
+    max: 65535
+  # SELinux
+  seLinux:
+    # SELinux is unsed in CaaSP
+    rule: 'RunAsAny'
+---
 kind: ClusterRole
 apiVersion: rbac.authorization.k8s.io/v1beta1
 metadata:
   name: flannel
 rules:
+  - apiGroups: ['extensions']
+    resources: ['podsecuritypolicies']
```

Comment:
PSP is still beta(?). Nevertheless I presume this will be a paper-over if we run a backward compatible kubernetes without PSP reconciler. Which may mean that we run quite underprivileged. Do we want to document the compatibility?

Reply:
Removing the beta I end up with:

```
Error from server (BadRequest): error when creating "STDIN": PodSecurityPolicy in version "v1" cannot be handled as a PodSecurityPolicy: no kind "PodSecurityPolicy" is registered for version "extensions/v1"
```

I've also noticed that you are using a really old kubernetes version when it comes to testing:
```
K8S_VERSION="${K8S_VERSION:-1.7.6} ---> ./dist/functional-test-k8s.sh
```
As a result I had to bump the version in the e2e tests to the latest stable one (1.13.2). However this up to you, I don't want to be too intrusive with this PR, just raise the security concerns.

Reply:
I suspect we need SYS_ADMIN because we mount /run and write to it (though I would agree with #1090 that we should be more specific). But a real test will prove it.
I am in favour of bumping e2e to updated kube version (PR please?).

This PR is good, but we need some hint about the compatibility - what would be the oldest supported Kube version with this merged in?